server/client.go

```diff
@@ -1233,19 +1233,27 @@ func (c *client) traceMsg(msg []byte) {
 	}
 }
 
+// Traces an incoming operation.
+// Will NOT check if tracing is enabled, does NOT need the client lock.
 func (c *client) traceInOp(op string, arg []byte) {
 	c.traceOp("<<- %s", op, arg)
 }
 
+// Traces an outgoing operation.
+// Will check if tracing is enabled, DOES need the client lock.
+func (c *client) traceOutOpIfOk(op string, arg []byte) {
```

Comment:
I wonder if we really need the Ok not Not OK version. Currently there is not many places where you call the "Ok" versions..

Reply:
That's why I didn't add one for traceInOp and traceOp.

Reply:
I'll try to push trace into the caller. Not every trace is right at the beginning.
So I'll have to check how many are left and how much code duplication this brings.
If it seems reasonable. to do, I'll see how many *IfOk are left and possibly remove them.